src/web/app/pages-instructor/instructor-course-enroll-page/instructor-course-enroll-page.component.ts

```diff
@@ -122,58 +125,99 @@ export class InstructorCourseEnrollPageComponent implements OnInit {
+        this.remainingStudentChunks = this.remainingStudentChunks.slice(1, this.remainingStudentChunks.length);
+      },
+      complete: () => {
+        this.showEnrollResults = true;
+        this.statusMessage.pop(); // removes any existing error status message
+        this.statusMessageService.showSuccessToast('Enrollment successful. Summary given below.');
+
+        const allStudentEntries: StudentEnrollRequest[] = [];
+        this.allStudentChunks.forEach((chunk: StudentEnrollRequest[]) => allStudentEntries.push(...chunk));
+
+        this.enrollResultPanelList =
+            this.populateEnrollResultPanelList(this.existingStudents, enrolledStudents, allStudentEntries);
+
+        this.studentService.getStudentsFromCourse({ courseId: this.courseId }).subscribe((resp: Students) => {
+          this.existingStudents = resp.students;
+          if (!this.isExistingStudentsPanelCollapsed) {
+            const existingStudentTable: Handsontable = this.hotRegisterer.getInstance(this.existingStudentsHOT);
+            this.loadExistingStudentsData(existingStudentTable, this.existingStudents);
+          }
+          this.isExistingStudentsPresent = true;
         });
+      },
+      error: (resp: ErrorMessageOutput) => {
+        this.enrollErrorMessage = resp.error.message;
+      },
+    });
   }
 
+  private handleTimeoutForEnrollRequest(prevRequests: Observable<Students>): Observable<Students> {
+
+    return prevRequests.pipe(finalize(() => this.isEnrolling = false))
+        .pipe(takeWhile(() => this.remainingStudentChunks.length > 0))
+        .pipe(catchError(() => {
+          // First request in the list caused the timeout
+          const currentLongRequest: StudentEnrollRequest[] = this.remainingStudentChunks[0];
+
+          // Break the long request into half
+          const studentEnrollRequest1: StudentEnrollRequest[] =
+              currentLongRequest.slice(0, currentLongRequest.length / 2);
+
+          const studentEnrollRequest2: StudentEnrollRequest[] =
+              currentLongRequest.slice(currentLongRequest.length / 2, currentLongRequest.length);
+
+          this.remainingStudentChunks =
+              this.remainingStudentChunks.filter((chunk: StudentEnrollRequest[]) =>
+                  chunk !== currentLongRequest);
+          this.remainingStudentChunks =
+              [studentEnrollRequest1, studentEnrollRequest2, ...this.remainingStudentChunks];
+
+          // Prepare chunks for new stream
+          const newRequests: Observable<Students> = concat(
+              ...this.remainingStudentChunks.map((studentChunk: StudentEnrollRequest[]) => {
+                const studentsEnrollRequest: StudentsEnrollRequest = {
+                  studentEnrollRequests: studentChunk,
+                };
+                return this.studentService.enrollStudents(
+                    this.courseId, studentsEnrollRequest,
+                );
+              }),
+          );
+
+          // Return a new stream
```

Comment:
Is this comment necessary? From the function signature, we kind of can tell.
The evil of comment is that sometimes it might lie; if the dev changes the code and forgets to change the comment accordingly, the comment is effectively not only useless, but instead confusing.